### PR TITLE
fix: `extract_struct_from_enum_variant` assist should resolve Self generic arg

### PR DESCRIPTION
**Pull request by Young-Flash**
close https://github.com/rust-lang/rust-analyzer/issues/16197

### CODE REVIEW

**Comment by Young-Flash** [comment]
sorry I miss the case that nested generic arg with `Self`, corrected now.

it's ready for review.

File: crates/ide-assists/src/handlers/extract_struct_from_enum_variant.rs [hunk 1]
```diff
@@ -246,6 +246,29 @@ fn create_struct_def(
...
+                                    Some(ted::replace(it, type_arg.syntax()))
+                                } else {
+                                    None
+                                }
+                            })
+                            .count(),
+                    ),
+                    _ => None,
+                })
+                .count();
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/extract_struct_from_enum_variant.rs:270]: This should use the `PathTransform::transform_path` function to transform the `Self` path

**Review by Veykril — CHANGES_REQUESTED** [review]

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
Thanks!
@bors r+

**Pull request merged**